quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java

```diff
@@ -387,16 +420,24 @@ private static String resolveLogOutput(String value, ConfigSourceInterceptorCont
     /**
      * Add tracing info to the log if the format is not explicitly set, and tracing and {@code includeTraceOption} options are enabled
      */
-    private static String addTracingInfo(String value, Option<Boolean> includeTraceOption) {
+    private static String addTracingAndMdcInfo(String value, Option<Boolean> includeTraceOption, Option<Boolean> includeMdcOption) {
         var isTracingEnabled = TracingPropertyMappers.isTracingEnabled();
         var includeTrace = isTrue(includeTraceOption);
         var isChangedLogFormat = !DEFAULT_LOG_FORMAT.equals(value);
+        var includeMdc = isTrue(includeMdcOption);
 
-        if (!isTracingEnabled || !includeTrace || isChangedLogFormat) {
+        if (isChangedLogFormat) {
             return value;
         }
```

Comment:
nitpicking, I would check the log format right at the beggining of the method before checking what is enabled or not

```
        if (!DEFAULT_LOG_FORMAT.equals(value)) {
            return value;
        }
        var isTracingEnabled = TracingPropertyMappers.isTracingEnabled();
        var includeTrace = isTrue(includeTraceOption);
        var includeMdc = isTrue(includeMdcOption);
```